packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -0,0 +1,77 @@
+
+The column definition will also be responsible for specifying the appropriate sort function to use based on the type of data it knows is associated with `operationalDataFieldName`. The sort function will be specified by a `sortFunction` getter on the column, that returns a `TableColumnSortFunction` value. `TableColumnSortFunction` is a new enum that will be introduced that defines the set of sort functions that can be applied to a column. Initially, the enum will consist of the value `basic`, which will correspond to [TanStack's `basic` sort function](https://tanstack.com/table/v8/docs/api/features/sorting#sorting-functions), and the value `localeAwareCaseSensitive`, which will be backed by a function written within nimble. That function will leverage [localeCompare()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) to perform the sort.
+
+For performance reasons, the table will not support custom sort functions on a column. Additionally, a column can only be sorted based on a single field within the table's data. These requirements guard against performance degradations from inefficient custom sort functions.
+
+#### Summary of new attributes on the base table column
+
+| attribute name | type                                                                                                         | default value                   | description                                                                                                                 |
+| -------------- | ------------------------------------------------------------------------------------------------------------ | ------------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
+| sort-direction | `TableColumnSortDirection`, defined as `{none: undefined, ascending: 'ascending', descending: 'descending'}` | `TableColumnSortDirection.none` | The direction the column is sorted                                                                                          |
+| sort-index     | `number` or `null` \*                                                                                        | `null`                          | The index for sorting the column. When multiple columns are sorted, they will be sorted from lowest index to highest index. |
```

Comment:
> sorted from lowest index to highest index

Some edge cases we can test when assigned as attributes "-Infinity" "-2" "-0" "3.14" "1e6" "Infinity" "NaN"

Reply:
I made a note in the spec that we should test with these edge cases.